test/functional/specs/C14409.js

```diff
@@ -0,0 +1,106 @@
+import { t, ClientFunction } from "testcafe";
+import createNetworkLogger from "../helpers/networkLogger";
+import getResponseBody from "../helpers/networkLogger/getResponseBody";
+import fixtureFactory from "../helpers/fixtureFactory";
+import cookies from "../helpers/cookies";
+import alloyEvent from "../helpers/alloyEvent";
+import getConsentCookieName from "../helpers/getConsentCookieName";
+import createResponse from "../../../src/core/createResponse";
+
+const networkLogger = createNetworkLogger();
+
+fixtureFactory({
+  title: "C14409 - Consenting to no purposes should be persisted.",
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+
+test.meta({
+  ID: "C14409",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const setConsentToOut = ClientFunction(() => {
+  return window.alloy("setConsent", { general: "out" });
+});
+
+test("C14409 - Consenting to no purposes should be persisted.", async () => {
+  const imsOrgId = "53A16ACB5CC1D3760A495C99@AdobeOrg";
+  await cookies.clear();
+
+  const configure = await alloyEvent("configure", {
+    configId: "9999999",
+    orgId: imsOrgId,
+    debugEnabled: true,
+    idMigrationEnabled: false
+  });
+
+  await configure.promise;
+
+  // send alloy event
+  const event1 = await alloyEvent({
+    viewStart: true
+  });
+
+  // apply user consent
+  await setConsentToOut();
+
+  await event1.promise;
+
+  const cookieName = getConsentCookieName(imsOrgId);
+
+  const consentCheck = await cookies.get(cookieName);
+
+  await t.expect(consentCheck).eql("general=out");
+
+  // test konductor response
+  await t.expect(networkLogger.edgeEndpointLogs.requests.length).eql(1);
+  await t
+    .expect(networkLogger.edgeEndpointLogs.requests[0].response.statusCode)
+    .eql(200);
+
+  const request = JSON.parse(
+    getResponseBody(networkLogger.edgeEndpointLogs.requests[0])
+  );
+
+  // read state:store handles from response (i.e. 'set a cookie')
+  await t.expect("handle" in request).ok();
+  await t.expect(request.handle.length).gt(0);
+
+  const storePayloads = createResponse(request).getPayloadsByType(
+    "state:store"
+  );
+  const cookiesToSet = storePayloads.reduce((memo, storePayload) => {
+    memo[storePayload.key] = storePayload;
+    return memo;
+  }, {});
```

Comment:
Hey @alshaw, thanks for making these updates. We probably should have written down our decisions or concrete steps after our team discussion. I believe we decided that we weren't going to inspect the response payload or look at the cookies at all because they are implementation details only. This means you can remove the code I've highlighted here and any of the cookie checking below. I think it makes sense to remove the first `alloyEvent` in this test too. Here's how I see the test going:

1. Configure Alloy (you don't need to set default consent--by default it is "in").
1. Set consent to "out".
1. Reload the page.
1. Configure Alloy (you don't need to set default consent--by default it is "in").
1. Try to send an event.
1. Check that the promise from the command is rejected and check the error message like you did below to make sure it was due to the user being opted out.

Similarly, for C14407, I see it going like this:

1. Configure alloy with default consent set to "pending".
1. Set consent to "in".
1. Reload the page.
1. Configure alloy with default consent set to "pending".
1. Try to send an event.
1. Check that the promise from the command is resolved.

For both tests, I don't think there's any need to check the actual networks requests, network responses, or cookies. Does that make sense? I'm happy to work through this with you if you'd like. Just hit me up on Slack.

Reply:
Ah okay, I misunderstood. I'll update C14407 and C14409 to reflect these steps.

Reply:
Thanks @alshaw! C14409 looks good. It looks like C14407 still needs to be updated though.